### PR TITLE
feat: Add parquet file detection

### PR DESCRIPTION
**Pull request by kwkelly**
Adds parquet file detection.

See [docs](https://parquet.apache.org/docs/file-format/) for specification

### CODE REVIEW

File: tree.go [hunk 1]
```diff
@@ -258,4 +258,5 @@ var (
 	xfdf    = newMIME("application/vnd.adobe.xfdf", ".xfdf", magic.Xfdf)
 	glb     = newMIME("model/gltf-binary", ".glb", magic.Glb)
 	jxr     = newMIME("image/jxr", ".jxr", magic.Jxr).alias("image/vnd.ms-photo")
+	parquet = newMIME("application/vnd.apache.parquet", ".parquet", magic.Par1)
```
Comment (kwkelly) [inline tree.go:261]: mime-type specification: https://www.iana.org/assignments/media-types/application/vnd.apache.parquet

Comment (gabriel-vasile) [inline tree.go:261]: `application/vnd.apache.parquet` was standardised at IANA a few months ago. `application/x-parquet` was the pseudo-standard before that and some software might still be using the old mime, ex: https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-processing-job-data-format-tabular.html#clarify-processing-job-data-format-tabular-prereq

To accommodate for older software let's add `application/x-parquet` as alias.
```suggestion
	parquet = newMIME("application/vnd.apache.parquet", ".parquet", magic.Par1).
		alias("application/x-parquet")
```

**Review by gabriel-vasile — CHANGES_REQUESTED** [review]
Thank you, the PR looks good, just one comment.

File: tree.go (hunk 1 above)
Comment (kwkelly) [reply tree.go]: Branch updated to include `application/x-parquet` alias.

**Review by gabriel-vasile — APPROVED** [review]

**Pull request merged**